src/App.jsx

```diff
@@ -14,45 +20,81 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [select, setSelect] = useState('');
+
+  const sortMethod = (howToSort) => {
+    switch (howToSort) {
+      case ALPHABETICALLY:
+        setGoods([...goods].sort((a, b) => a.localeCompare(b)));
+        setSelect(ALPHABETICALLY);
+        break;
+      case LENGTH:
+        setGoods([...goods].sort((a, b) => a.length - b.length));
+        setSelect(LENGTH);
+        break;
+      case REVERSE:
+        setGoods([...goods].reverse());
+        setSelect(REVERSE);
+        break;
+      default:
+        setGoods(goodsFromServer);
+        setSelect('');
+    }
+  };
+
+  const Reset = () => {
+    if (select !== '') {
+      return (
+        <button
+          type="button"
+          className={`button is-danger ${select !== RESET ? 'is-light' : undefined}`}
+          onClick={() => sortMethod()}
+        >
+          Reset
+        </button>
+      );
+    }
+
+    return null;
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${select !== ALPHABETICALLY ? 'is-light' : undefined}`}
```

Comment:
use classnames library for dynamic render classes
this library alreade exist in project
https://www.npmjs.com/package/classnames

```
import cn from 'classnames'
```
```suggestion
          className={cn('button is-info',  { 'is-light': select !== ALPHABETICALLY })}
```